specs/phase1/beacon-chain.md

```diff
@@ -530,33 +533,75 @@ def get_indexed_attestation(beacon_state: BeaconState, attestation: Attestation)
     )
 ```
 
+#### `get_committee_count_delta`
+
+```python
+def get_committee_count_delta(state: BeaconState, start_slot: Slot, stop_slot: Slot) -> uint64:
+    """
+    Return the sum of committee counts between ``[start_slot, stop_slot)``.
+    """
+    committee_sum = 0
+    for slot in range(start_slot, stop_slot):
+        count = get_committee_count_at_slot(state, Slot(slot))
+        committee_sum += count
+    return committee_sum
+```
+
 #### `get_start_shard`
 
 ```python
 def get_start_shard(state: BeaconState, slot: Slot) -> Shard:
-    # TODO: implement start shard logic
-    return Shard(0)
+    """
+    Return the start shard at ``slot``.
+    """
+    current_epoch_start_slot = compute_start_slot_at_epoch(get_current_epoch(state))
+    active_shard_count = get_active_shard_count(state)
+    if current_epoch_start_slot == slot:
+        return state.current_epoch_start_shard
+    elif current_epoch_start_slot > slot:
+        # Current epoch or the next epoch lookahead
+        shard_delta = get_committee_count_delta(state, start_slot=current_epoch_start_slot, stop_slot=slot)
+        return Shard((state.current_epoch_start_shard + shard_delta) % active_shard_count)
+    else:
+        # Previous epoch
+        shard_delta = get_committee_count_delta(state, start_slot=slot, stop_slot=current_epoch_start_slot)
+        max_committees_per_epoch = MAX_COMMITTEES_PER_SLOT * SLOTS_PER_EPOCH
```

Comment:
Optional 😅   In Prysm, I ended up doing 👇
```python
max_shards_per_epoch = MAX_COMMITTEES_PER_SLOT * SLOTS_PER_EPOCH * active_shard_count
return Shard(
            # Ensure positive
            (state.current_epoch_start_shard + max_shards_per_epoch - shard_delta)
            % active_shard_count
        )
```

Reply:
If we do so, the variable name `max_shards_per_epoch` will need to be changed. Any suggestion? :)

Reply:
I see. On second though, I don't feel strongly about this one. We can leave it as it is